http-clients/java-nio-client/src/main/java/software/amazon/awssdk/http/nio/java/internal/ListToByteBufferProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.java.internal;
+
+import io.reactivex.Flowable;
+import io.reactivex.processors.PublishProcessor;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+@SdkInternalApi
+public class ListToByteBufferProcessor implements Subscriber<List<ByteBuffer>> {
+
+
+    private PublishProcessor<List<ByteBuffer>> processor;
+    private Flowable<ByteBuffer> publisherToSdk;
+
+    private final CompletableFuture<Void> terminated = new CompletableFuture<>();
+
+
+    ListToByteBufferProcessor() {
+        processor = PublishProcessor.create();
+        publisherToSdk = processor.map(list -> {
+            int listSize = list.stream().mapToInt(Buffer::capacity).sum();
+            ByteBuffer processedByteBuffer = ByteBuffer.allocate(listSize);
+            list.forEach(processedByteBuffer::put);
+            return processedByteBuffer;
+        });
+    }
+
+    Flowable<ByteBuffer> getPublisherToSdk() {
```

Comment:
generally `public` methods are first, so can you move this and `getTerminated` to the bottom after all the public methods?